### DISCUSSION TITLE
Is it possible to turn off (power down) the robot vacuum to save energy?

### DISCUSSION THREAD

**Discussion by Archetrix** [Hypfer/Valetudo#1851] [Q&A / Support]
I'm looking for a way to remotely (i.e. MQTT) send my Valetudo powered Roborock S5 series vacuum a command to power down. I know i can do this by removing it from the dock first (or power down the dock) and then press and hold the power button on the vacuum. But i'd like to do this with my home control. 

I had the dock on a switched power socket for a while and had a schedule to power on the socket at 07:00. A daily cleaning starts at 09:00 and during that the automation watches the battery level to go below 80% first and then wait until it reached 100% again to turn the dock off. 

The rest of the day the robot would then use up available charge in standby. And i figured out that would be a bad idea for longer because it would reduce the lifespan of the battery. Not worth it to save a few Watt-hours.

A better approach would be to turn off the robot as well preserving the charge. And then i would let the charge go up to 80% only and turn off the robot and dock for the rest of the day and night. Then next morning charge the rest before the cleaning schedule and by turning on the dock it would wake up the robot.

If i wanted to use the robot for a spot cleaning i could "wake" it by powering up the dock any time of the day - even remotely. Maybe i would add a little ESP Board to the dock to do the power switching internally :)

I did not find a means within the "Swagger UI". The Robot or System API's have got no "PUT"
Wouldn't this be a nice feature to have?

**Comment by Hypfer** [comment]
As you've already discovered, this is not possible as none of the robots that are currently supported offer a way of fully shutting down the robot from the linux environment. While you _can_ run the `shutdown` command, that will only shut down the linux side, keeping the rest of the robot awake. There's also no way of waking it from that state again.

As Valetudo is just a cloud replacement and not a custom firmware, there's nothing it can do about that.

**Reply by Archetrix** [reply]
Hi @Hypfer to me the shutdown command accessible via API would be the solution. If i figured out how to include it into your codebase would you be willing to honor a pull request for this?

I'd look into the settings as well to explicitly enable a shutdown via MQTT.

**Reply by Hypfer** [reply]
> While you can run the shutdown command, that **will only shut down the linux side, keeping the rest of the robot awake**. There's also **no way of waking it from that state** again.
